### PR TITLE
fix:  Deleted room is displayed in the 'Assign task'

### PR DESCRIPTION
**Pull request by nkdengineer**
### Fixed Issues

$ https://github.com/Expensify/App/issues/50418
PROPOSAL: https://github.com/Expensify/App/issues/50418#issuecomment-2400036186

### Tests

1. Create a workspace
2. Add some messages in the #admin channel
3. Click on FAB
4. Click on create task
5. Search for the #admin room of the created workspace
6. Create the task
7. Verify the task is created in the #admin room
8. Delete the workspace
9. Click on FAB on the LHN
10. Click on assign task
11. Verify that: After deleting a workspace, it's rooms are not displayed in the account

- [x] Verify that no errors appear in the JS console

### Offline tests

same as above
### QA Steps

same as above
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/dafc7c0d-bb67-4689-9e50-f14dc8e307dd

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/ba05956e-3efe-4b54-b9ae-7cb5200d24d2

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/66295d4d-98de-41a0-b340-c96857698ca5

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/bbe6fa46-8bce-40d0-a2b1-681f458bea67

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/f46d2fbf-6b73-40ad-9b2a-041a43b8faf3

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/8a644d33-029b-4c27-8b1b-c606dc1e70f7

</details>

### CODE REVIEW

**Comment by nkdengineer** [comment]
@Ollyws i fixed this bug

**Review by Ollyws — APPROVED** [review]
LGTM.

**Review by tgolen — CHANGES_REQUESTED** [review]
Can you please add a unit test to ensure that the bug doesn't come back?

**Comment by tgolen** [comment]
Bump for unit tests, please.

**Comment by nkdengineer** [comment]
working on this PR

**Comment by nkdengineer** [comment]
@tgolen, I'm a bit confused about how we can add a unit test for this case.

**Comment by tgolen** [comment]
What's your confusion and how can I help resolve it?

**Comment by nkdengineer** [comment]
@tgolen My confusion is what test we can add in this case and where we can add this because it's not a util function and it's just a variable in the component.

**Comment by tgolen** [comment]
You should be able to write unit tests for components. I think the test should:

- Put some "setup" data into Onyx (eg. the "deleted room")
- Render the FAB component
- Verify that it doesn't show the deleted room

**Comment by nkdengineer** [comment]
@tgolen let me try

**Comment by nkdengineer** [comment]
@tgolen I think in this case we can create a util function to get `quickActionAvatars`, `quickActionTitle`, `hideQABSubtitle` then we can add a unit test for this util. what do you think?

**Comment by tgolen** [comment]
Sounds like that would work as well, yeah 👍 Good thinking

**Comment by nkdengineer** [comment]
@Ollyws i added, please check again

File: tests/unit/ReportUtilsTest.ts [hunk 1]
```diff
@@ -975,6 +975,22 @@ describe('ReportUtils', () => {
         });
     });
 
+    describe('getQuickActionDetails', () => {
```
Comment (tgolen) [inline tests/unit/ReportUtilsTest.ts:978]: Thanks for adding this test!

Can you please follow [this format](https://github.com/Expensify/App/blob/4833a6c1cb39db9034843230ffab5455a112f1e2/tests/README.md#documenting-tests) to add a little bit of context and documentation to the test?

**Review by tgolen — CHANGES_REQUESTED** [review]

File: tests/unit/ReportUtilsTest.ts (hunk 1 above)
Comment (nkdengineer) [reply tests/unit/ReportUtilsTest.ts:978]: @tgolen i updated

Comment (tgolen) [reply tests/unit/ReportUtilsTest.ts:978]: OK, close enough, but that format is meant to have each comment start with literal "Given", "When" and "Then" words :D

**Review by tgolen — APPROVED** [review]

File: tests/unit/ReportUtilsTest.ts (hunk 1 above)
Comment (nkdengineer) [reply tests/unit/ReportUtilsTest.ts:978]: Noted for the feature.

**Comment by kavimuru** [comment]
Failing with original bug https://github.com/Expensify/App/issues/50418  in offline mode in web

https://platform.applause.com/services/links/v1/external/c82a81c1ead07a5322da74736c935190a41520378ecfd893ab5d44488d38d669

**Pull request merged**